### ISSUE TITLE
How can I use large batch_size when facing OOM during pre-training RoBERTa?

### ISSUE THREAD

**Issue by luciencho** [pytorch/fairseq#925]
In the paper, 8K batch size and 512 sequence length are used for pre-training RoBERTa. Does mixed precision training play an import role during training?

**Comment by myleott** [comment]
You can use `--update-freq` to accumulate gradients from multiple batches to simulate arbitrarily larger batch sizes. For example, if you're training on 4 GPUs and set `--update-freq=2`, then it's equivalent to training on 8 GPUs.

Mixed precision training (`--fp16`) does help reduce memory usage slightly, but the most straightforward way to train with large batches is to distribute them across many workers. At times we trained on 1024 GPUs (i.e., 8 sequences per GPU, `--update-freq=1`).

Another interesting finding, which will make its way into a future preprint, is that it's better to train BERT_base with 2K sequences compared to 256, even controlling for the compute requirements. I.e., 2K batch size converges faster than 256 batch size, even when trained for the same total number of epochs.

**Comment by lematt1991** [comment]
This seems to be resolved.  Closing for now.  Please open another issue if you are still having problems.  Thanks!
